src/compiler.rkt

```diff
@@ -81,8 +80,18 @@
   (define compile
     (make-compiler 'ival
                    #:input->value (lambda (prog _) (ival (bf prog)))
-                   #:op->procedure (lambda (op) (operator-info op 'ival))
-                   #:op->itypes (lambda (op) (operator-info op 'itype))
+                   #:op->procedure (lambda (op)
+                                    (match op
+                                      [(? repr-conv? impl) ; strangeness with specs
+                                       (operator-info (impl->operator impl) 'ival)]
```

Comment:
Aren't these always the identity anyway?

Reply:
I think this was because of our stricter separation of `operator-info` and `impl-info`. The IR issue is not resolved just quarantined to a few places like here. Not ideal.